membership-common/src/main/scala/com/gu/zuora/soap/Client.scala

```diff
@@ -81,49 +78,39 @@ class Client(
     Future.unit.flatMap(_ => request(action, Option(periodicAuth.get), reader))
 
-  def extendedAuthenticatedRequest[T <: Result](action: => Action[T])(implicit reader: Reader[T]): Future[T] =
+  def extendedAuthenticatedRequest[T <: Result](action: => Action[T])(implicit reader: Reader[T], logPrefix: LogPrefix): Future[T] =
     Future.unit.flatMap(_ => request(action, Option(periodicAuth.get), reader, extendedHttpClient))
 
-  def query[T <: Query](where: String)(implicit reader: readers.Query[T]): Future[Seq[T]] =
+  def query[T <: Query](where: String)(implicit reader: readers.Query[T], logPrefix: LogPrefix): Future[Seq[T]] =
     authenticatedRequest(Actions.Query(reader.format(where))).map { case QueryResult(results) => reader.read(results) }
 
-  def query[T <: Query](where: ZuoraFilter)(implicit reader: readers.Query[T]): Future[Seq[T]] =
-    query(where.toFilterString)(reader)
+  def query[T <: Query](where: ZuoraFilter)(implicit reader: readers.Query[T], logPrefix: LogPrefix): Future[Seq[T]] =
+    query(where.toFilterString)(reader, logPrefix)
 
-  def queryOne[T <: Query](where: String)(implicit reader: readers.Query[T]): Future[T] =
-    query(where)(reader).map(
+  def queryOne[T <: Query](where: String)(implicit reader: readers.Query[T], logPrefix: LogPrefix): Future[T] =
+    query(where)(reader, logPrefix).map(
       _.headOption
         .getOrElse(throw new ZuoraQueryException(s"Query '${reader.getClass.getSimpleName} $where' returned 0 results, expected one")),
     )
 
-  def queryOne[T <: Query](where: ZuoraFilter)(implicit reader: readers.Query[T]): Future[T] =
-    queryOne(where.toFilterString)(reader)
-
-  def parent[P <: Query, C <: Query](child: C, foreignKey: C => String)(implicit reader: readers.Query[P]): Future[P] =
-    queryOne[P](parentFilter(child, foreignKey))(reader)
-
-  def child[P <: Query with Identifiable, C <: Query](parent: P)(implicit reader: readers.Query[C]): Future[C] =
-    queryOne[C](childFilter(parent))(reader)
+  def queryOne[T <: Query](where: ZuoraFilter)(implicit reader: readers.Query[T], logPrefix: LogPrefix): Future[T] =
+    queryOne(where.toFilterString)(reader, logPrefix)
 
 }
 
 object Client {
-  implicit class ChildOps[C <: Query](child: C) {
-    def parent[P <: Query](foreignKey: C => String)(implicit client: Client, reader: readers.Query[P]) =
-      client.parent(child, foreignKey)(reader)
-  }
-
-  implicit class ParentOps(parent: Query with Identifiable) {
-    def child[C <: Query](implicit client: Client, reader: readers.Query[C], ct: ClassTag[C]) =
-      client.child(parent)(reader)
-  }
 
   def parentFilter[C <: Query, P <: Query](child: C, foreignKey: (C) => String): SimpleFilter =
     SimpleFilter("Id", foreignKey(child))
 
   def childFilter[C <: Query, P <: Query with Identifiable](parent: P): SimpleFilter =
     SimpleFilter(parent.objectName + "Id", parent.id)
+
+  val noLogPrefix: LogPrefix = new LogPrefix {
+    override def message: String = "no-id"
```

Comment:
for SOAP calls there's usually an identity id, except for health checks or scheduled jobs.  So I enforced by default in the compiler - rather than using the "NoPrefix" one I made a dummy prefix to pass in.